test/metabase/query_processor_test/expression_aggregations_test.clj

```diff
@@ -377,3 +378,82 @@
                     mt/rows
                     (map second)
                     (map sort))))))))
+
+(defmethod driver/database-supports? [::driver/driver ::expression-aggregation-literals]
+  [driver _feature database]
+  (and (driver/database-supports? driver :expressions database)
+       (driver/database-supports? driver :expression-aggregations database)))
+
+;; literal expressions in aggregations not yet supported for these drivers
+(doseq [driver [:mongo :oracle :redshift :sqlite :sqlserver :vertica]]
+  (defmethod driver/database-supports? [driver ::expression-aggregation-literals]
+    [_driver _feature _database]
+    false))
```

Comment:
Mentioning specific drivers by name in tests is a major code smell.

If these drivers support `:expressions` and `:expression-aggregations` but _not_ `::expression-aggregation-literals`, then it sounds like this is a **new feature** that should have its own flag, set to true on only those drivers that legitimately support it!

Reply:
Yeah, agreed. Previously there was TODO comment here about turning these into a real DB feature, but then I backtracked thinking it "shouldn't" be required, but as you say the failing tests are a good indication that it probably is required in practice.

Reply:
How this is going to be used? The FE will allow literals in all cases (as it does now).

Reply:
Well current master won't let you create a custom column with a top-level literal value, right?

But actually I have the same question about how this will work outside of tests. I thought the answer was going to be that drivers that don't support a particular feature raise an error if they're asked to compile something they don't support, but I only found one example of that in the mongo driver so far.

Reply:
@appleby `lib/diagnose-expression` could return an error when top-level literals are used for a database that doesn't support them. And, if we do that, we could merge our changes without fixing all the issues at once.

Reply:
Probably makes sense to do that anyway, since it's better than letting the user input the expression and fail when the query runs, even if we raise an explicit error.

Up to @bshepherdson if he's ok to merge this to the integration branch without the driver feature change. If so, I can do it in a follow on PR when I'm back next week.

Reply:
I've marked it ready for review. If the tests pass and you want merge to unblock other work in the integration branch, go for it. Otherwise, I'll take a look at the driver feature on Monday.

Reply:
Fixed in latest updates. Added a new `:expression-literals` driver feature.